From f87050dafef3022a3f95f8518c6defa88d9fb3de Mon Sep 17 00:00:00 2001
From: cchaduvu <chandrasekhar.chaduvula@amd.com>
Date: Fri, 17 Apr 2026 16:49:07 +0530
Subject: [PATCH 2/2] max96716: update nDes6 init for 12-bit 8MP sensor VCID
 support

- Enable 12-bit data path configuration for 8MP sensors
- Update CSI/PHY register settings for 12-bit output
- Correct VC-ID programming for multi-stream 8MP use case
- Align Link-A and Link-B register settings for consistent VC handling
- Remove legacy and unused VC/format register programming

This change adds proper 12-bit and VC-ID support for
8MP sensors on the MAX96716-based FMC design.

Signed-off-by: cchaduvu <chandrasekhar.chaduvula@amd.com>
---
 .../src/usr_sensor/include/fmc/max96716.h     | 64 +++++++------------
 1 file changed, 24 insertions(+), 40 deletions(-)

diff --git a/sensor_drivers/src/usr_sensor/include/fmc/max96716.h b/sensor_drivers/src/usr_sensor/include/fmc/max96716.h
index a0e8c9f..0a7326c 100755
--- a/sensor_drivers/src/usr_sensor/include/fmc/max96716.h
+++ b/sensor_drivers/src/usr_sensor/include/fmc/max96716.h
@@ -365,52 +365,36 @@ RegI2CT nDes5_init[] = {
 RegI2CT nDes6_init[] = {
 	{MAX929X_TABLE_WAIT, MAX929X_TABLE_WAIT_MS},
 	{0x0161, 0x31},
-	{0x0050, 0x00},
-	{0x0051, 0x01},
-	{0x0052, 0x02},
-	{0x0053, 0x03},
-	{0x0320, 0x2f},
-	{0x0323, 0x20},
-	{0x0316, 0xAC},
-	{0x0317, 0xBB},
-	{0x0318, 0xB0},
-	{0x0319, 0x6A},
-	{0x0313, 0x62},
-	{0x031A, 0x30},
-	{0x0333, 0x4E}, 
-
+	{0x0320, 0x2f},   //0x37 for 2250
+	{0x0323, 0x2f},
+	{0x0316, 0x80},
+	{0x0317, 0xBC},
+	{0x0318, 0x00},
+	{0x0319, 0x6C},
+	
+	{0x0333, 0x4E},
 	{0x040A, 0xD0},
 	{0x044A, 0xD0},
 	{0x048A, 0xD0},
 	{0x04CA, 0xD0},
-	{0x044B, 0x07}, //LinkA
-	{0x046D, 0x15}, //LinkA
-	{0x044D, 0x2B}, //LinkA
-	{0x044E, 0x2B}, //LinkA
-	{0x044F, 0x00}, //LinkA
-	{0x0450, 0x00}, //LinkA
-	{0x0451, 0x01}, //LinkA
-	{0x0452, 0x01}, //LinkA
-	
-//Changing VC-ID to 2 for these Link-B registers
-	{0x048B, 0x3F},
-	{0x04AD, 0x55},
-	{0x04AE, 0x05},
-	{0x048D, 0x2B},
-	{0x048E, 0xAB},
+
+	{0x044B, 0x07},
+	{0x046D, 0x15},
+	{0x044D, 0x2C},
+	{0x044E, 0x2C},
+	{0x044F, 0x00},
+	{0x0450, 0x00},
+	{0x0451, 0x01},
+	{0x0452, 0x01},
+
+	{0x048B, 0x07},
+	{0x04AD, 0x15},
+	{0x048D, 0x2C},
+	{0x048E, 0x6C},
 	{0x048F, 0x00},
-	{0x0490, 0x80},
+	{0x0490, 0x40},
 	{0x0491, 0x01},
-	{0x0492, 0x81},
-
-	// {0x048B, 0x07},
-	// {0x04AD, 0x15},
-	{0x0493, 0x6B},
-	{0x0494, 0xEB},
-	{0x0495, 0x40},
-	{0x0496, 0xC0},
-	{0x0497, 0x41},
-	{0x0498, 0xC1},
+	{0x0492, 0x41},
 	{MAX929X_TABLE_WAIT, MAX929X_TABLE_WAIT_MS},
 	{MAX929X_TABLE_END, 0}
 };
-- 
2.43.0

